tests/test_toa_flag.py

```diff
@@ -50,3 +52,112 @@ def test_flag_value_str(self):
+ uwl_191012_013412.0000_0000.rf.pTF 1102.00000000 58768.08628513784793768 16.29100 pks
+ uwl_191013_013912.0000_0000.rf.pTF 1102.00000000 58769.08628494704621659 18.55400 pks
+ uwl_191014_011651.0000_0000.rf.pTF 1102.00000000 58770.06840417512788122 21.57900 pks
+ uwl_191015_063631.0000_0000.rf.pTF 1102.00000000 58771.28385332518061546 23.07400 pks
+ uwl_191029_231442.0000_0000.rf.pTF 1102.00000000 58785.98420216310909581 18.68600 pks
+ uwl_191111_020722.0000_0000.rf.pTF 1102.00000000 58798.10659743353399165 17.39200 pks
+ uwl_191128_000422.0000_0000.rf.pTF 1102.00000000 58815.01493200936268835 20.28800 pks
+ uwl_191207_024252.0000_0000.rf.pTF 1102.00000000 58824.11892727732292840 25.10700 pks
+"""
+    t = toa.get_TOAs(io.StringIO(s))
+    t.set_flag_values("test", 1)
+    for i in range(t.ntoas):
+        assert t["flags"][i]["test"] == "1"
+
+
+def test_flag_set_array():
+    s = """FORMAT 1
+MODE 1
+ uwl_191009_015952.0000_0000.rf.pTF 1102.00000000 58765.09374976854768846 22.04600 pks
+ uwl_191010_000912.0000_0000.rf.pTF 1102.00000000 58766.02430580012498496 23.74000 pks
+ uwl_191012_013412.0000_0000.rf.pTF 1102.00000000 58768.08628513784793768 16.29100 pks
+ uwl_191013_013912.0000_0000.rf.pTF 1102.00000000 58769.08628494704621659 18.55400 pks
+ uwl_191014_011651.0000_0000.rf.pTF 1102.00000000 58770.06840417512788122 21.57900 pks
+ uwl_191015_063631.0000_0000.rf.pTF 1102.00000000 58771.28385332518061546 23.07400 pks
+ uwl_191029_231442.0000_0000.rf.pTF 1102.00000000 58785.98420216310909581 18.68600 pks
+ uwl_191111_020722.0000_0000.rf.pTF 1102.00000000 58798.10659743353399165 17.39200 pks
+ uwl_191128_000422.0000_0000.rf.pTF 1102.00000000 58815.01493200936268835 20.28800 pks
+ uwl_191207_024252.0000_0000.rf.pTF 1102.00000000 58824.11892727732292840 25.10700 pks
+"""
+    t = toa.get_TOAs(io.StringIO(s))
+    t.set_flag_values("test", np.ones(len(t)))
+    for i in range(t.ntoas):
+        assert float(t["flags"][i]["test"]) == 1
+
+
+def test_flag_set_partialarray():
+    s = """FORMAT 1
+MODE 1
+ uwl_191009_015952.0000_0000.rf.pTF 1102.00000000 58765.09374976854768846 22.04600 pks
+ uwl_191010_000912.0000_0000.rf.pTF 1102.00000000 58766.02430580012498496 23.74000 pks
+ uwl_191012_013412.0000_0000.rf.pTF 1102.00000000 58768.08628513784793768 16.29100 pks
+ uwl_191013_013912.0000_0000.rf.pTF 1102.00000000 58769.08628494704621659 18.55400 pks
+ uwl_191014_011651.0000_0000.rf.pTF 1102.00000000 58770.06840417512788122 21.57900 pks
+ uwl_191015_063631.0000_0000.rf.pTF 1102.00000000 58771.28385332518061546 23.07400 pks
+ uwl_191029_231442.0000_0000.rf.pTF 1102.00000000 58785.98420216310909581 18.68600 pks
+ uwl_191111_020722.0000_0000.rf.pTF 1102.00000000 58798.10659743353399165 17.39200 pks
+ uwl_191128_000422.0000_0000.rf.pTF 1102.00000000 58815.01493200936268835 20.28800 pks
+ uwl_191207_024252.0000_0000.rf.pTF 1102.00000000 58824.11892727732292840 25.10700 pks
+"""
+    t = toa.get_TOAs(io.StringIO(s))
```

Comment:
If all these TOAs objects can be identical, perhaps they could be created in a fixture?

Reply:
yes, done.